src/peer-queue.js

```diff
@@ -17,34 +18,22 @@ class PeerQueue {
    * Create from a given peer id.
    *
    * @param {PeerId} id
-   * @param {function(Error, PeerQueue)} callback
-   * @returns {void}
+   * @returns {Promise<PeerQueue>}
    */
-  static fromPeerId (id, callback) {
-    utils.convertPeerId(id, (err, key) => {
-      if (err) {
-        return callback(err)
-      }
-
-      callback(null, new PeerQueue(key))
-    })
+  static async fromPeerId (id) {
+    const key = await promisify(cb => utils.convertPeerId(id, cb))()
```

Comment:
This should throw now right? Should we add `@throws` to the jsdocs? It will make it easier for anyone consuming to know that they should ensure they're handling it. If so, we should add this to the other methods that also throw.

Reply:
@jacobheun  Thats correct. What is the correct way to document throws for async functions?

Reply:
Looking more through the jsdocs repo there's not really a correct way, unfortunately. Ideally we'd be able to do something like this, https://github.com/jsdoc/jsdoc/issues/1467, to be explicit. 

For now, either way we have to do some level of assumption looking at the docs. Since the function is async, a throw is just a promise rejection. So we either need to catch it, or `.catch` the promise.

I suppose it makes sense to leave it out for now until the docs can be explicit.